app/presenters/document_export_presenter.rb

```diff
@@ -46,6 +47,13 @@ def complete_images_hash(edition_images)
     end
   end
 
+  def provide_doctype_information(edition, output)
+    output[:document_type_information] = edition.public_send(:news_article_type).as_json if edition.respond_to?(:news_article_type)
```

Comment:
This doesn't need to use `public_send`

Reply:
Yet you do above in the example snippet you gave above:

`output[type] = edition.public_send(type).key if edition.respond_to?(type)`

May I ask why not here? Should I be using .send()? I mean this is being replaced anyway but I'll need to learn somehow.

Reply:
[#public_send](https://ruby-doc.org/core-2.5.2/Object.html#method-i-public_send) is used to call a particular method on the object. Thus it makes sense to use if we are iterating through a list of methods to call or constructing the method name dynamically.

In your code it was being called without any dynamism which meant it was equivalent to just calling the method. For example: `edition.public_send(:news_article_type).as_json ==  edition.news_article_type.as_json`.

Reply:
Ah understood, thanks.